### PR TITLE
feat(components): overlay functionality on conditionals - I233

### LINKED ISSUE #233: Conditional variables UI
**Is your feature request related to a problem? Please describe.**
As a user I would like to enable and disable conditional variables, and to switch between two options in conditional variables.

**Describe the solution you'd like**
See this Loom video explaining the UI:
https://www.loom.com/share/694e7927a744475a98b1cf81154c3673

Smart Clause with three conditional variables, marked with red text
https://projects.invisionapp.com/d/main#/console/16741009/394631841/preview
image

User hovers over Smart Clause to see there is an additional conditional variable that can be enabled. Signified by a plus icon.
https://projects.invisionapp.com/d/main#/console/16741009/394631840/preview
image

User hovers over icon to see what text will appear when they click on it.
https://projects.invisionapp.com/d/main#/console/16741009/394631839/preview
image

The user clicked on the text, making the conditional variable text appear.
https://projects.invisionapp.com/d/main#/console/16741009/394631838/preview
image

The user is hovering over conditional variable text. A tooltip tells the user the text will be hidden if they click
https://projects.invisionapp.com/d/main#/console/16741009/394631837/preview
image

The user has clicked to hide the text and it is hidden again
https://projects.invisionapp.com/d/main#/console/16741009/394631840/preview
image

If the user hovers on a conditional variable that switches between two text options, they will see a message like this. When they click, the text will change to the other option.
https://projects.invisionapp.com/d/main#/console/16741009/394633329/preview
image

### PR DESCRIPTION
**Pull request by irmerk**
# Issue #233
Closes #233 with full support of `boolean` and `if`/`else` conditional variables in the UI

### Changes
- Create an addition symbol component
  - Tooltip which reads on hover _"Hide text"_ when true and _"Change to: `TEXT`"_ when false
- Wrap conditionals with tooltip
  - Tooltip which reads  _"Show text: `OTHER_TEXT`"_ on hover
- Rely on Slate key for all functionality
- Store conditional information on state in clause IFF conditional is an empty string

### Flags
- ~Issue with this approach is shown in the demo~
  - ~Conditionals that span more than one line break the logic~

### Related Issues
- Pull Request #288

### CODE REVIEW

File: src/components/Conditional.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    whenFalse: node.data.get('whenFalse'),
+    isFalse: node.text === node.data.get('whenFalse'),
+  };
+
+  const toggleConditional = (key) => {
+    const newInlineJSON = {
+      object: 'inline',
+      type: 'conditional',
+      data: {
+        id: 'forceMajeure',
```
Comment (jeromesimeon) [inline src/components/Conditional.js]: Is this `id` hardcoded? Should it be based on the variable name?

Comment (irmerk) [reply src/components/Conditional.js]: Nice very good catch!

**Comment by irmerk** [comment]
> Also wondering if it wouldn't be better to try and unify the conditional and conditional+else code and UX. Maybe something for further discussion.

Could you expand on this? Not sure I understand.

**Comment by jeromesimeon** [comment]
> > Also wondering if it wouldn't be better to try and unify the conditional and conditional+else code and UX. Maybe something for further discussion.
> 
> Could you expand on this? Not sure I understand.

Maybe I followed wrong but it seemed like there was two components one `ConditionalBoolean` and one `ConditionalSwitch` ?

_Looking again, I see that they cover different part of the logic so I'm probably just completely mistaken_

**Comment by irmerk** [comment]
> Maybe I followed wrong but it seemed like there was two components one `ConditionalBoolean` and one `ConditionalSwitch` ?
> 
> _Looking again, I see that they cover different part of the logic so I'm probably just completely mistaken_

Ah yes yes yes, so ideally the `ConditionalPlugin` would house all the logic. However, the hovering over a clause and thus displaying editing/deleting/testing icons is in the `ClausePlugin`, which that `ConditionalBoolean` needs to display. So I had to unfortunately separate them.

**Pull request merged**